src/plugins/numberline/components/editable-numberline-value.tsx

```diff
@@ -0,0 +1,103 @@
+import { observer } from 'mobx-react';
+import React, { useState, useRef, useEffect } from 'react';
+
+import "./numberline-tile.scss";
+
+interface IEditableValueProps {
+  readOnly?: boolean;
+  isTileSelected: boolean;
+  value: number;
+  offset: number;
+  minOrMax: "min" | "max";
+  onValueChange: (newValue: number) => void;
+}
+
+export const EditableNumberlineValue: React.FC<IEditableValueProps> = observer(function NumberlineTile(props) {
+  const { readOnly, isTileSelected, value, offset, minOrMax, onValueChange } = props;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.select();
+    }
+  }, [isEditing]);
+
+  const handleClick = () => {
+    if (!readOnly && !isEditing) {
+      setIsEditing(true);
+      // Check if the input element exists before focusing and selecting
+      if (inputRef.current) {
+        inputRef.current.select();
+      }
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    let inputField = undefined;
+    let numberEntered = undefined;
+    const { key } = e;
+    switch (key) {
+      case "Enter":
+        inputField = (e.target as HTMLInputElement).value;
+        if (checkIfNumber(inputField)){
+          numberEntered = Number(inputField);
+          onValueChange(numberEntered);
+        }
+        setIsEditing(false);
+        break;
+      case "Escape":
+      case "Tab":
+        setIsEditing(false);
+        break;
+    }
+  };
+
+  const checkIfNumber = (input: string): boolean => {
+    const result = Number(input);
+    const isNumeric = !isNaN(result) && isFinite(result);
+    return isNumeric;
+  };
+
+  //----------------------- Determine Styling for Border Box -----------------------
+
+  const borderBoxOffset = `${offset + 4}px`;
+  const hideBorderAndResetBackground = !isTileSelected;
+  const borderBoxPositionProperty = minOrMax === "min" ? { left: borderBoxOffset } : { right: borderBoxOffset };
+  const borderBoxBorderProperty = hideBorderAndResetBackground ? { border: "none" } : { border: "1.5px solid #949494" };
+  const borderBoxBackgroundProperty = hideBorderAndResetBackground ? { backgroundColor: "white" }
+                                                                   : { backgroundColor: "#f0f9fb" };
+  const borderBoxStyle = { ...borderBoxPositionProperty, ...borderBoxBorderProperty, ...borderBoxBackgroundProperty };
+
+  return (
+    <div className="border-box" style={borderBoxStyle} onClick={handleClick}>
```

Comment:
You should use a class based on `isTileSelected` to set the `border` and `background-color` here rather than inline styling. It's ok to use inline styling for `left` and `right` because that's dynamically determined based on props (if you could make this static in the css that would be better, but I don't know how easy that would be).